### ISSUE TITLE
OnPR hung in main during EmitC tests - batchnorm training op?

### ISSUE THREAD

**Issue by jameszianxuTT** [tenstorrent/tt-mlir#5633]
Attempting to repro locally.

**Comment by jameszianxuTT** [comment]
Repro candidate from emitc.sh build + test script:
```
// build script
llvm-lit -sv test/ttmlir/EmitC/TTNN
python tools/ttnn-standalone/ci_compile_dylib.py

// test script
llvm-lit build/test/ttmlir/EmitC/TTNN 
ttrt emitc build/test/ttmlir/EmitC/TTNN --flatbuffer build/test/ttmlir/EmitC/TTNN
```

**Comment by jameszianxuTT** [comment]
Could be batch norm op hanging due to tt-metal bug https://github.com/tenstorrent/tt-mlir/issues/5601

Local repro shows hang at 
```
2025-10-31 20:44:56,033 - INFO - test command for binary=build/test/ttmlir/EmitC/TTNN/batch_norm/batch_norm_training.mlir.ttnn is: /opt/ttmlir-toolchain/venv/bin/ttrt run build/test/ttmlir/EmitC/TTNN/batch_norm/batch_norm_training.mlir.ttnn --program-index all --loops 1 --save-artifacts  --artifact-dir /localdev/jameszianxu/tt-mlir/ttrt-artifacts 
2025-10-31 20:44:57,683 - INFO - set log file=
2025-10-31 20:44:57,683 - INFO - setting artifacts folder path=/localdev/jameszianxu/tt-mlir/ttrt-artifacts
```
